attribute-projection-registry/src/main/java/org/hypertrace/core/attribute/service/projection/AttributeProjectionRegistry.java

```diff
@@ -0,0 +1,30 @@
+package org.hypertrace.core.attribute.service.projection;
+
+import static org.hypertrace.core.attribute.service.v1.ProjectionOperator.PROJECTION_OPERATOR_CONCAT;
+import static org.hypertrace.core.attribute.service.v1.ProjectionOperator.PROJECTION_OPERATOR_HASH;
+
+import java.util.Map;
+import java.util.Optional;
+import org.hypertrace.core.attribute.service.projection.functions.Concatenate;
+import org.hypertrace.core.attribute.service.projection.functions.Hash;
+import org.hypertrace.core.attribute.service.v1.AttributeKind;
+import org.hypertrace.core.attribute.service.v1.ProjectionOperator;
+
+public class AttributeProjectionRegistry {
+
+  private static final Map<ProjectionOperator, AttributeProjection> PROJECTION_MAP =
+      Map.of(
+          PROJECTION_OPERATOR_CONCAT,
```

Comment:
I think we need to be able to chain the projections also right? For example, HASH(CONCAT(a, b)) so that we get the id automatically. That can come in later but it's better to keep them in mind.

Reply:
Talked offline, but summary: yes we need and do support that, it's the responsibility of the layer above to resolve the arguments to literal values, which may include recursive projections.